### PR TITLE
Fix label renaming regression

### PR DESCRIPTION
**Pull request by wheresrhys**
The github api now expects name changes to be sent as `new_name` not `name` (a new label is created if name is used)

Have reported the possible regression to github, but it feels like a conscious change to the labels model by github, and they've perhaps been a bit careless about the breaking nature of the change

This PR contains a sketch fix which definitely works, though may need tweaking to avoid side-effects I'm not aware of

### CODE REVIEW

File: lib/github-label-api.js [hunk 1]
```diff
@@ -62,6 +62,17 @@ class ApiClient {
 		labelName = encodeURIComponent(labelName);
 		return new Promise((resolve, reject) => {
 			const endpoint = `/repos/${repo}/labels/${labelName}`;
+			
+			const newLabel = {...label}
```
Comment (JakeChampion) [inline lib/github-label-api.js:66]: Let's change this from spread syntax to Object.assign, that way our listing configuration will still work

Comment (wheresrhys) [reply lib/github-label-api.js]: Ah you meant 'linting'

Confusing typo :joy:

File: lib/github-label-api.js [hunk 2]
```diff
@@ -62,6 +62,17 @@ class ApiClient {
 		labelName = encodeURIComponent(labelName);
 		return new Promise((resolve, reject) => {
 			const endpoint = `/repos/${repo}/labels/${labelName}`;
+
+			const newLabel = Object.assign({}, label);
```
Comment (JakeChampion) [inline lib/github-label-api.js:66]: I don't see where `newLabel` is being used, is it meant to be used on line 76 - in the `this.apiClient.patch(...)` call?

Comment (wheresrhys) [reply lib/github-label-api.js:66]: 🤦

Comment (wheresrhys) [reply lib/github-label-api.js:66]: Was wondering why no tests needed updating. Addressed now

**Review by JakeChampion — APPROVED** [review]

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v2.0.1 :tada:

**Pull request merged**